### PR TITLE
feat: GCS setup and modification of the URLField to ImageField

### PR DESCRIPTION
**Pull request by nataniadeandra**
**Summary of Changes:**

1. **Google Cloud Storage Setup:** Integrated the backend with Google Cloud Storage. This was achieved by adding the `dotted-lexicon-415505-86934bc0f37c.json` file, which contains the necessary credentials, and updating the settings with the appropriate credentials and bucket name.

2. **Photo Model Update:** Modified the Photo model in `venue/models.py`. The field previously using `URLField` has been changed to `ImageField`. This adjustment is crucial for the integration with Google Cloud Storage. The URLs in Google Cloud Storage are dynamic and can change over time. By managing the image references directly in the backend, we ensure that the frontend always receives valid and current image URLs, thereby reducing the risk of broken links.

3. **Test Modification:** Aligned the tests in `venue/tests.py` with the changes made in the Photo model. This ensures that our tests remain valid and continue to provide valuable feedback on the functionality of our application.

4. **Image Addition for Testing:** Added an image file named `empathymap.jpg` for testing purposes. This image will be used to validate the functionality of the updated Photo model and the integration with Google Cloud Storage.

5. **Test Case Refactoring:** Refactored the base test cases for the Venue and Photo models in `venue/tests.py`. The tests for these two models are now separate, making them clearer and easier to read. This change improves the maintainability of our tests and makes it easier to identify and fix issues in the future.

### CODE REVIEW

File: dotted-lexicon-415505-86934bc0f37c.json
Comment (emiriko) [inline dotted-lexicon-415505-86934bc0f37c.json:1]: Personally, I think this needs to not be pushed to github, as our repository is public and this file contains sensitive information. What does everyone think?

**Review by catherineeangel — APPROVED** [review]
Nice, by implementing GCS directly it will ease the storage consumption of our app. Overall the code looks good and the tests covered all the possible cases

**Review by emiriko — COMMENTED** [review]
I really like your approaches and refactoring; it looks good and efficient. Keep up the good work!

File: dotted-lexicon-415505-86934bc0f37c.json
Comment (catherineeangel) [reply dotted-lexicon-415505-86934bc0f37c.json:1]: Right, are there any other way to do so? can we put the filename as an enviroment variable and refer it from the .env ?

Comment (JohannesSetiawan) [reply dotted-lexicon-415505-86934bc0f37c.json:1]: Yeah this is a very sensitive information, should be put on the environment variables.

**Review by JohannesSetiawan — CHANGES_REQUESTED** [review]
Everything looks good except the sensitive information in the .json. Should be disclosed. Even better if you can revert to commit to that .json so it wont appear in the commit history.

File: dotted-lexicon-415505-86934bc0f37c.json
Comment (nataniadeandra) [reply dotted-lexicon-415505-86934bc0f37c.json:1]: I completely understand everyone's perspectives. I acknowledge that I may have overlooked the fact that storing sensitive data in GitHub secrets is not a recommended practice.

@catherineeangel @JohannesSetiawan, your idea to use GitHub environment variables and generate the `.json` file during the build process is excellent. Will revert and implement all of your suggestions. Thank you. ☺️

**Review by swastikanata — APPROVED** [review]
Agree with others above, everything should be fine after you hide the private information with environment secrets.Other than that, look good to me.

**Comment by nataniadeandra** [comment]
Hello, I have moved all the necessary credentials for Google Cloud Storage from the earlier files to environment variables in Github and Vercel. I kindly request those who have previously commented or requested changes to either approve this pull request or provide additional feedback if necessary.

@swastikanata @catherineeangel @JohannesSetiawan @emiriko @carlenee

**Review by JohannesSetiawan — APPROVED** [review]

**Review by JohannesSetiawan — APPROVED** [review]
Good. The problem is fixed.

**Pull request merged**